arbitrator/prover/src/machine.rs

```diff
@@ -1358,6 +1358,12 @@ impl Machine {
             .unwrap_or(true)
     }
 
+    pub fn next_instruction_is_read_hotshot(&self) -> bool {
+        self.get_next_instruction()
+            .map(|i| i.opcode == Opcode::ReadHotShotCommitment)
+            .unwrap_or(true)
```

Comment:
Do you know why we default to `true` here?

Reply:
Yeah it is quite confusing here. This function is similar with `next_instruction_is_hostio`.
When there is no next instruction, which means this is already the last step in this machine, it will hit `unwrap_or` case.
And this function is `mainly` used to stops the machine at a certain position (return true) and it should stop of course at the last step.
I know it is not a good way to write this, but I think it would be easily understood if we keep it similar to the existed one